tower_cli/cli/misc.py

```diff
@@ -204,3 +205,30 @@ def config(key=None, value=None, scope='user', global_=False, unset=False):
             UserWarning
             )
     click.echo('Configuration updated successfully.')
+
+
+@click.command()
+@click.argument('username', required=True)
+@click.option('--password', required=True, prompt=True, hide_input=True)
+def login(username, password):
+    """
+    Retrieves and stores an OAuth2 personal auth token.
+    """
+    if not supports_oauth():
+        raise exc.TowerCLIError(
+            'This version of Tower does not support OAuth2.0'
+        )
+
+    # Explicitly set a basic auth header for PAT acquisition (so that we don't
+    # try to auth w/ an existing user+pass or oauth token in a config file)
+    r = client.post(
+        '/users/{}/personal_tokens/'.format(username),
+        data={"description": "Tower CLI", "application": None, "scope": "read"},
+        headers={'Authorization': 'Basic %s' % base64.b64encode(':'.join([
+            username, password
+        ]))}
+    )
+
+    if r.status_code == 201:
+        token = r.json()['token']
+        config.main(['oauth_token', token, '--scope=user'])
```

Comment:
Well this is a very interesting approach.

Reply:
```shell
~/dev/tower-cli tower-cli config

# User options (set with `tower-cli config`; stored in ~/.tower_cli.cfg).
host: tower.example.org:8043

# Defaults.
username:
use_token: False
verbose: False
certificate:
format: human
color: True
description_on: False
verify_ssl: True
oauth_token:
password:

~/dev/tower-cli tower-cli inventory list --tower-oauth-token=PB3X8L2hXjN4mxdYxt7dd0PbInhSDk
== ============== ============
id      name      organization
== ============== ============
 1 Demo Inventory            1
== ============== ============
```